Sources/swift-bundler/Commands/RunCommand.swift

```diff
@@ -137,6 +139,59 @@ struct RunCommand: AsyncCommand {
               ExampleCommand("swift bundler simulators list")
             }.show()
 
+            Foundation.exit(1)
+          }
+        }
+
+      case .visionOSSimulator:
+        if let searchTerm = simulatorSearchTerm {
+          // Get matching simulators
+          let simulators = try SimulatorManager.listAvailableSimulators(searchTerm: searchTerm)
+            .unwrap().sorted { first, second in
+              // Put booted simulators first and sort by name length
+              if first.state == .shutdown, second.state == .booted {
+                return false
+              } else if first.name.count > second.name.count {
+                return false
+              } else {
+                return true
+              }
+            }
+
+          guard let simulator = simulators.first else {
+            log.error(
+              "Search term '\(searchTerm)' did not match any simulators. To list available simulators, use the following command:"
+            )
+
+            Output {
+              ""
+              ExampleCommand("swift bundler simulators list")
+            }.show()
+
+            Foundation.exit(1)
+          }
+
+          if simulators.count > 1 {
+            log.info("Found multiple matching simulators, using '\(simulator.name)'")
+          }
+
+          return .visionOSSimulator(id: simulator.id)
+        } else {
+          let allSimulators = try SimulatorManager.listAvailableSimulators().unwrap()
+
+          // If an visionOS simulator is booted, use that
+          if allSimulators.contains(where: { $0.state == .booted }) {
+            return .visionOSSimulator(id: "booted")
+          } else {
+            // swiftlint:disable:next line_length
+            log.error(
+              "To run on the visionOS simulator, you must either use the '--simulator' option or have a valid simulator running already. To list available simulators, use the following command:"
+            )
+
+            Output {
+              ExampleCommand("swift bundler simulators list")
+            }.show()
+
             Foundation.exit(1)
           }
```

Comment:
Avoid all this code duplication by combining this code with the iOSSimulator case. Also please add a `// TODO:` comment to remind me to refactor that code into a separate function a some point 😅 That looks well overdue.